app/src/main/java/com/github/se/eventradar/ui/qrCode/QrCodeFriendUi.kt

```diff
@@ -0,0 +1,165 @@
+package com.github.se.eventradar.ui.qrCode
+
+import android.widget.Toast
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Tab
+import androidx.compose.material3.TabRow
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ConstraintLayout
+import com.github.se.eventradar.R
+import com.github.se.eventradar.ui.BottomNavigationMenu
+import com.github.se.eventradar.ui.navigation.NavigationActions
+import com.github.se.eventradar.ui.navigation.Route
+import com.github.se.eventradar.ui.navigation.TOP_LEVEL_DESTINATIONS
+import com.github.se.eventradar.ui.navigation.TopLevelDestination
+import com.github.se.eventradar.viewmodel.qrCode.NavigationEvent
+import com.github.se.eventradar.viewmodel.qrCode.QrCodeFriendViewModel
+
+@Composable
+fun QrCodeScreen(viewModel: QrCodeFriendViewModel, navigationActions: NavigationActions) {
+  val navigateState by viewModel.navigationEvent.collectAsState()
+
+  // React to changes in navigation state
+  LaunchedEffect(navigateState) {
```

Comment:
This is not necessary. You can simply navigate right where needed in the logic below using `navigationActuions.navigate({route})`

Reply:
DONE, just not going to screen desired as not implemented yet